### ISSUE TITLE
Accept input from stdin?

### ISSUE THREAD

**Issue by prashcr** [palantir/tslint#1590]
I'm trying to get tslint working with a vim linter, however to do this I need to be able to pass input via stdin. 
https://github.com/w0rp/ale/issues/20

To my knowledge, all major JS linters support this feature (jshint, eslint, jscs).

**Comment by prashcr** [comment]
Ended up using a bash wrapper that buffers stdin to a temp file.

Would still be great if it supported stdin out of the box and I wouldn't mind helping out with that.

**Comment by adidahiya** [comment]
we'd accept a PR for this, but it might get a little complex due to the interactions with the compiler APIs.

**Comment by prashcr** [comment]
Thanks!

Will try to get something in over the next month or two.

**Comment by rjmunro** [comment]
I thought that it was working, as when I typed `tslint - < filename.ts` I didn't get any errors, but it looks like the accpetance of `-` as a filename is a bug.

**Comment by ianks** [comment]
@prashcr This would be such an excellent feature! I am currently trying to configure https://github.com/sbdchd/neoformat to use tslint and am stumbling across some problems which this would solve.

**Comment by prashcr** [comment]
@ianks I agree!
Unfortunately I've been busy as of late, but I still see myself working on this at sometime soon.

**Comment by tusharmath** [comment]
@adidahiya Do you think we can add a special flag to take input via stdin? For Eg:

```bash
$ git diff --name-only | tslint --fix --stdin
```

**Comment by ajafff** [comment]
@tusharmath for your example you could simply use `xargs`:

```bash
git diff --name-only | xargs tslint --fix
```

**Comment by tusharmath** [comment]
@ajafff Thanks a ton! You saved me a day :P  BTW what's the use case of `stdin` in that case?

**Comment by ExplodingCabbage** [comment]
@tusharmath *one* use case (surely not the only one) is linting the output of `git show :somefile` (which outputs the staged form of `:somefile`) in a commit hook, so that if I've *staged* a linting error, then fixed it but not staged the fix, the hook will still fail when I try to commit.

**Comment by w0rp** [comment]
I would like to add that in addition to supporting `--stdin`, a `--stdin-filename` option should be supported, like in ESLint. This allows linter plugins to pass the contents of text buffers in place of filenames, so you can spot errors before a file is saved.

You essentially replace `tslint whatever.ts` with `tslint --stdin --stdin-filename whatever.ts` and send the contents of the buffer to TSLint via stdin.

**Comment by l5x** [comment]
This would be really helpful :+1:

**Comment by w0rp** [comment]
I'm going to give this a go. I'll create a pull request with the first thing I can think of at some point, and keep editing it until everything actually works.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
